pkg/controllers/background/controller.go

```diff
@@ -0,0 +1,135 @@
+	kyvernov1beta1 "github.com/kyverno/kyverno/api/kyverno/v1beta1"
+	kyvernoclient "github.com/kyverno/kyverno/pkg/client/clientset/versioned"
+	kyvernov1informers "github.com/kyverno/kyverno/pkg/client/informers/externalversions/kyverno/v1"
+	kyvernov1beta1informers "github.com/kyverno/kyverno/pkg/client/informers/externalversions/kyverno/v1beta1"
+	kyvernov1listers "github.com/kyverno/kyverno/pkg/client/listers/kyverno/v1"
+	kyvernov1beta1listers "github.com/kyverno/kyverno/pkg/client/listers/kyverno/v1beta1"
+	"github.com/kyverno/kyverno/pkg/config"
+	"github.com/kyverno/kyverno/pkg/dclient"
+	controllerutils "github.com/kyverno/kyverno/pkg/utils/controller"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const (
+	maxRetries = 10
+	workers    = 3
+)
+
+type controller struct {
+	// clients
+	client        dclient.Interface
+	kyvernoClient kyvernoclient.Interface
+
+	// listers
+	cpolLister kyvernov1listers.ClusterPolicyLister
+	polLister  kyvernov1listers.PolicyLister
+	urLister   kyvernov1beta1listers.UpdateRequestNamespaceLister
+
+	// queue
+	queue workqueue.RateLimitingInterface
+}
+
+func NewController(
+	client dclient.Interface,
+	kyvernoClient kyvernoclient.Interface,
+	cpolInformer kyvernov1informers.ClusterPolicyInformer,
+	polInformer kyvernov1informers.PolicyInformer,
+	urInformer kyvernov1beta1informers.UpdateRequestInformer,
+) *controller {
+	c := controller{
+		client:        client,
+		kyvernoClient: kyvernoClient,
+		cpolLister:    cpolInformer.Lister(),
+		polLister:     polInformer.Lister(),
+		urLister:      urInformer.Lister().UpdateRequests(config.KyvernoNamespace()),
+		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "background-controller"),
+	}
+	controllerutils.AddDefaultEventHandlers(logger, cpolInformer.Informer(), c.queue)
+	controllerutils.AddDefaultEventHandlers(logger, polInformer.Informer(), c.queue)
+	return &c
+}
+
+func (c *controller) Run(stopCh <-chan struct{}) {
+	controllerutils.Run(logger, c.queue, workers, maxRetries, c.reconcile, stopCh)
+}
+
+func (c *controller) reconcile(key, namespace, name string) error {
+	logger := logger.WithValues("key", key, "namespace", namespace, "name", name)
+	logger.Info("reconciling ...")
+	_, err := c.loadPolicy(namespace, name)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return c.cleanupPolicy(logger, key)
+		}
+		return err
+	}
+	return nil
+}
+
+func (c *controller) loadPolicy(namespace, name string) (kyvernov1.PolicyInterface, error) {
+	if namespace == "" {
+		return c.cpolLister.Get(name)
+	} else {
+		return c.polLister.Policies(namespace).Get(name)
+	}
+}
+
+func (c *controller) cleanupPolicy(logger logr.Logger, key string) error {
+	// generatePolicyWithClone := pkgCommon.ProcessDeletePolicyForCloneGenerateRule(p, c.client, c.kyvernoClient, c.urLister, p.GetName(), logger)
```

Comment:
What is the purpose of this ?

Reply:
It was used to enqueue UR for such deleted clone policies.
https://github.com/kyverno/kyverno/blob/24cdc59f78228f5830eec8503d24725faf3945db/pkg/background/generate/cleanup/controller.go#L128-L139

When a clone policy is deleted, Kyverno process the UR and cleans up target resource based on `synchronize` setting. 

@prateekpandey14 - per your previous comment in another PR, URs are deleted once it is successfully processed. Do we still need this logic to cleanup targets?

Reply:
Yep but the call to `pkgCommon.ProcessDeletePolicyForCloneGenerateRule` seems to remove a label from the source resource, do we need it ?
The same source resource can be used by multiple policies right ?

Reply:
>call to pkgCommon.ProcessDeletePolicyForCloneGenerateRule seems to remove a label from the source resource

Yes it remove the policy label from the source resource. i think policy label can be  used to list down the resource its related to. 

> The same source resource can be used by multiple policies right ?

Yes, then there will  be comma `,` separated policy names